metadata/src/main/java/edu/unc/lib/dl/validation/MODSValidator.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright 2008 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.validation;
+
+import static edu.unc.lib.dl.xml.JDOMNamespaceUtil.SCHEMATRON_VALIDATION_REPORT_NS;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.util.stream.Collectors;
+
+import javax.xml.transform.stream.StreamSource;
+import javax.xml.validation.Schema;
+
+import org.apache.commons.io.IOUtils;
+import org.jdom2.Document;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.SAXException;
+
+import edu.unc.lib.dl.schematron.SchematronValidator;
+
+/**
+ * Asserts that MODS complies with the XSD and controlled vocabularies,
+ * as per the configured schematron validator.
+ *
+ * @author bbpennel
+ * @author harring
+ *
+ */
+public class MODSValidator {
+    private static final Logger log = LoggerFactory.getLogger(MODSValidator.class);
+
+    private SchematronValidator schematronValidator = null;
+    private Schema modsSchema = null;
+
+    public SchematronValidator getSchematronValidator() {
+        return schematronValidator;
+    }
+
+    public void setSchematronValidator(SchematronValidator schematronValidator) {
+        this.schematronValidator = schematronValidator;
+    }
+
+    public Schema getModsSchema() {
+        return modsSchema;
+    }
+
+    public void setModsSchema(Schema modsSchema) {
+        this.modsSchema = modsSchema;
+    }
+
+    /**
+     * Validates a MODS description from a File object
+     *
+     * @param file
+     * @throws IOException
+     */
+    public InputStream validate(File file) throws IOException {
+        return validate(new ByteArrayInputStream(Files.readAllBytes(file.toPath())));
+    }
+
+    /**
+     * Validates a MODS description. NB: this method only supports InputStream types that can be reset
+     * for multiple reads. Other types will be converted to a ByteArrayInputStream prior to validation.
+     *
+     * @param docStream
+     * @throws MetadataValidationException
+     * @throws IOException
+     */
+    public InputStream validate(InputStream docStream) throws MetadataValidationException, IOException {
+        if (!docStream.markSupported()) {
```

Comment:
Looking at this now, I feel like returning an inputstream from a validation method is kind of confusing, in part because its supporting one case.  It would seem like the client calling the validator would have a better idea of what kind of data it was receiving and how to handle its datastream?  What would you think of having it instead throw an IllegalArgumentException if the inputstream doesn't support `mark` instead?

As a part of that, the javadoc should list that exception and when it would be thrown.

Alternatively, both the validate methods need to state what they are returning.